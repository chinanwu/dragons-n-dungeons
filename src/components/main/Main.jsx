import React from 'react';

import Ability from './Ability.jsx';
import Character from './Character.jsx';
import Combat from './Combat.jsx';
import { ABILITY_NAMES } from '../../constants/Abilities';

import './Main.less';

export const Main = () => {
  return (
    <div id="main">
      <div>
        <Ability
          id={ABILITY_NAMES['str'].toLowerCase()}
          name={ABILITY_NAMES['str']}
          className="Ability"
        />
        <Ability
          id={ABILITY_NAMES['con'].toLowerCase()}
          name={ABILITY_NAMES['con']}
          className="Ability"
        />
        <Ability
          id={ABILITY_NAMES['dex'].toLowerCase()}
          name={ABILITY_NAMES['dex']}
          className="Ability"
        />
      </div>
      <div>
        <Character />
        <Combat />
      </div>
      <div>
        <Ability
          id={ABILITY_NAMES['wis'].toLowerCase()}
          name={ABILITY_NAMES['wis']}
        />
        <Ability
          id={ABILITY_NAMES['int'].toLowerCase()}
          name={ABILITY_NAMES['int']}
        />
        <Ability
          id={ABILITY_NAMES['cha'].toLowerCase()}
          name={ABILITY_NAMES['cha']}
        />
      </div>
    </div>
  );
};

export default Main;
