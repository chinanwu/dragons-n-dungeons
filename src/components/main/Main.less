#main {
  display: flex;
  justify-content: space-between;
}

.Ability {
  margin-bottom: 20px;
}
