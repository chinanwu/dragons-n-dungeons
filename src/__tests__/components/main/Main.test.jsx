import { shallow } from 'enzyme';
import React from 'react';

import { Main } from '../../../components/main/Main.jsx';

jest.unmock('../../../components/main/Main.jsx');

describe('Main component', () => {
  describe('rendering', () => {
    it('renders Main', () => {
      const wrapper = shallow(<Main />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
