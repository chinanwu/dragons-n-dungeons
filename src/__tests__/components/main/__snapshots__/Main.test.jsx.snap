// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component rendering renders Main 1`] = `
<div
  id="main"
>
  <div>
    <Memo(Connect(Ability))
      id="strength"
      name="Strength"
    />
    <Memo(Connect(Ability))
      id="constitution"
      name="Constitution"
    />
    <Memo(Connect(Ability))
      id="dexterity"
      name="Dexterity"
    />
  </div>
  <div>
    <Memo(Connect(Character)) />
    <Memo(Connect(Combat)) />
  </div>
  <div>
    <Memo(Connect(Ability))
      id="wisdom"
      name="Wisdom"
    />
    <Memo(Connect(Ability))
      id="intelligence"
      name="Intelligence"
    />
    <Memo(Connect(Ability))
      id="charisma"
      name="Charisma"
    />
  </div>
</div>
`;
